src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_NuclearReactor.java

```diff
@@ -544,87 +524,139 @@ public void explodeMultiblock() {
-						MathUtils.roundToClosestInt(MathUtils.randDouble(1, 50)/10),
-						MathUtils.roundToClosestInt(MathUtils.randDouble(1, 50)/10)
-				};
-				final int fluorineContent = (100-outputChances[0]-outputChances[1]-outputChances[2]-outputChances[3]);
-				Logger.WARNING("Fluorine remaining: "+fluorineContent);
-				outputArrayOfGases = new FluidStack[]{
-						FLUORIDES.LITHIUM_FLUORIDE.getFluid(outputChances[0]),
-						FLUORIDES.NEPTUNIUM_HEXAFLUORIDE.getFluid(outputChances[1]),
-						FLUORIDES.TECHNETIUM_HEXAFLUORIDE.getFluid(outputChances[2]),
-						FLUORIDES.SELENIUM_HEXAFLUORIDE.getFluid(outputChances[3]),
-						Materials.Fluorine.getGas(fluorineContent)
-				};
+		}
+		if (aSpargeRecipe == null) {
+			return aOutputGases;
+		}		
+		int aSpargeGasAmount = aSpargeRecipe.mInputGas.amount;		
+		FluidStack depletionStack = aSpargeRecipe.mInputGas.copy();
+		depletionStack.amount = aSpargeGasAmount;
+		this.depleteInput(depletionStack);
+		updateSlots();		
+		for (int i=0;i<aSpargeRecipe.mFluidOutputs.length;i++) {
+			if (aSpargeRecipe.mInputGas.isFluidEqual(spargeGas)) {
+				continue; // Skip sparge gas
 			}
+			int aGasAmount = MathUtils.randInt(0, aSpargeRecipe.mMaxOutputQuantity[i]);
+			FluidStack aOutput = aSpargeRecipe.mFluidOutputs[i];
+			aSpargeGasAmount -= aGasAmount;
+			aOutputGases.add(new FluidStack(aOutput.getFluid(), aGasAmount));			
+		}
+		if (aSpargeGasAmount > 0) {
+			aOutputGases.add(new FluidStack(aSpargeRecipe.mInputGas.getFluid(), aSpargeGasAmount));	
 		}
-		return outputArrayOfGases;
+		return aOutputGases;
 	}
 
 	@Override
 	public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
-		//Add Power if active
-		if (aBaseMetaTileEntity.isActive()){
-			//this.getBaseMetaTileEntity().increaseStoredEnergyUnits(this.mEUt, false);
-			
-			if (this.mEfficiency >= 500){
-				this.boostEu = true;
-				this.turnCasingActive(true);
+		if (aBaseMetaTileEntity.getWorld().isRemote) {
+			if (aBaseMetaTileEntity.isActive()){
+				// Set casings active if we're warmed up.
+				if (this.mEfficiency == this.getMaxEfficiency(null)){
+					this.turnCasingActive(true);
+				}
+				else {
+					this.turnCasingActive(false);
+				}
 			}
 			else {
-				this.boostEu = false;
 				this.turnCasingActive(false);
 			}
-			
-			if (MathUtils.randInt(1, 200) == 1){
-				//Utils.LOG_INFO("Adding U233");
-				this.addOutput(ELEMENT.getInstance().URANIUM233.getFluid(MathUtils.randInt(1, 10)));
-			}
-			
-			if (this.mDynamoHatches != null) {
-				for (GT_MetaTileEntity_Hatch_Dynamo tHatch : this.mDynamoHatches) {
-					if (tHatch.mTier >= 5){
-						if (isValidMetaTileEntity(tHatch)){
-							tHatch.getBaseMetaTileEntity().increaseStoredEnergyUnits(this.mEUt, false);
-							//Utils.LOG_WARNING("Adding "+this.mEUt+"eu to internal storage of dynamo "+hatchNo+".");
-						}						
-					}
-				}
-			}
-			
 		}
 		else {
-			this.turnCasingActive(false);
-		}
+			// Try output some Uranium-233
```

Comment:
We should move this else block to onRunningTick for obvious reasons.

Reply:
Also, it needs to check if the machine is running with an actual recipe instead of just warming up.

Reply:
No, the texture change is done when it's warmed up, regardless of anything else. That's what it symbolizes.

Reply:
I mean, this will make it output some URANIUM233 even when the machine is turned off. This is onPostTick.